### PR TITLE
Fix part of #2294: Migrating to TextInputLayout for AddProfileActivity

### PR DESCRIPTION
**Pull request by FareesHussain**
## Explanation

Fix part of #2294 
Fix part of #962 
Migrating for `AddProfileActivity`

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

## ScreenShots
Username Portrait | Username Landscape
--- | ---
image | image
image | image
image | image

Pin Portrait | Pin Landscape
--- | ---
image | image
image | image
image | image

## Video

addProfileActivity

## Tests
**Espresso**
image

**Robolectric**
image

### CODE REVIEW

**Comment by anandwana001** [comment]
Same comment from other issue, add test screenshots.

File: app/src/main/res/layout-land/add_profile_activity.xml [hunk 1]
```diff
@@ -119,36 +127,53 @@
...
+            </com.google.android.material.textfield.TextInputLayout>
+
+            <com.google.android.material.textfield.TextInputLayout
+              android:id="@+id/add_profile_activity_confirm_pin"
+              style="@style/TextInputLayout"
               android:layout_width="match_parent"
               android:layout_height="wrap_content"
               android:layout_marginStart="28dp"
-              android:layout_marginTop="8dp"
+              android:layout_marginTop="24dp"
```
Comment (anandwana001) [inline app/src/main/res/layout-land/add_profile_activity.xml]: is the new margin according to the mocks?

File: app/src/main/res/layout-sw600dp-land/add_profile_activity.xml [hunk 2]
```diff
@@ -73,18 +72,27 @@
...
+            app:layout_constraintTop_toBottomOf="@+id/add_profile_activity_required_heading_text_view">
+
+            <com.google.android.material.textfield.TextInputEditText
+              android:id="@+id/add_profile_activity_user_name_edit_text"
+              style="@style/TextInputEditText"
+              android:layout_width="match_parent"
+              android:layout_height="wrap_content"
+              android:hint="@string/add_profile_input_name"
+              android:maxLength="36"
+              android:singleLine="true"
```
Comment (anandwana001) [inline app/src/main/res/layout-sw600dp-land/add_profile_activity.xml:94]: reformat and re-arrange.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @FareesHussain Doing a quick review
We need test cases covering pressing of IME action button.

**Comment by FareesHussain** [comment]
@anandwana001 I have a doubt regarding IME_ACTION in this activity
for username i've added IME_ACTION_DONE and pressing it closes the keyboard
for pin i've added IME_ACTION_NEXT and pressing it the focus goes to confirm pin
for confirm pin i've added IME_ACTION_DONE and pressing the creates a new profile


It looks good or do i have to make changes for the IME ACTIONS

File: app/src/main/res/layout-sw600dp-land/add_profile_activity.xml (hunk 2 above)
Comment (FareesHussain) [reply app/src/main/res/layout-sw600dp-land/add_profile_activity.xml:94]: Done

File: app/src/main/res/layout-land/add_profile_activity.xml (hunk 1 above)
Comment (FareesHussain) [reply app/src/main/res/layout-land/add_profile_activity.xml]: No, I'll revert that

Comment (FareesHussain) [reply app/src/main/res/layout-land/add_profile_activity.xml]: Done

**Comment by anandwana001** [comment]
> @anandwana001 I have a doubt regarding IME_ACTION in this activity
> for username i've added IME_ACTION_DONE and pressing it closes the keyboard
> for pin i've added IME_ACTION_NEXT and pressing it the focus goes to confirm pin
> for confirm pin i've added IME_ACTION_DONE and pressing the creates a new profile
> 
> It looks good or do i have to make changes for the IME ACTIONS

Defer to @rt4914 
Closing keyboard from name sounds ok to me but if the check is marked then we can shift the focus to pin from the name, else the keyboard will get close.

File: app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt [hunk 3]
```diff
@@ -112,43 +113,48 @@ class AddProfileActivityPresenter @Inject constructor(
...
-        pin?.let {
-          profileViewModel.inputPin.set(it.toString())
-          profileViewModel.pinErrorMsg.set("")
-          inputtedPin = pin.isNotEmpty()
-          setValidPin(binding)
-        }
+
+    binding.addProfileActivityPinEditText.onTextChanged { pin ->
+      pin?.let {
+        profileViewModel.inputPin.set(it.toString())
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt]: You don't need `toString()`

File: app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt [hunk 4]
```diff
@@ -112,43 +113,48 @@ class AddProfileActivityPresenter @Inject constructor(
...
-        confirmPin?.let {
-          profileViewModel.inputConfirmPin.set(it.toString())
-          profileViewModel.confirmPinErrorMsg.set("")
-          inputtedConfirmPin = confirmPin.isNotEmpty()
-          setValidPin(binding)
-        }
+
+    binding.addProfileActivityConfirmPinEditText.onTextChanged { confirmPin ->
+      confirmPin?.let {
+        profileViewModel.inputConfirmPin.set(it.toString())
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt]: You don't need `toString()`

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

File: app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt (hunk 4 above)
Comment (FareesHussain) [reply app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt]: Done

File: app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt (hunk 3 above)
Comment (FareesHussain) [reply app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt]: Done

**Comment by FareesHussain** [comment]
@rt4914 please re-run the workflow
image

**Comment by FareesHussain** [comment]
@rt4914 PTAL

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @FareesHussain

**Pull request merged**